src/main/java/com/hms_networks/americas/sc/extensions/historicaldata/HistoricalDataManager.java

```diff
@@ -57,6 +59,175 @@ public static void exportAllHistoricalToFile(
+   * @return EBD string
+   */
+  private static String prepareHistoricalFifoReadEBDString(
+      String startTime,
+      String endTime,
+      boolean includeTagGroupA,
+      boolean includeTagGroupB,
+      boolean includeTagGroupC,
+      boolean includeTagGroupD,
+      boolean stringHistorical) {
+
+    // Check for valid group selection
+    if (!includeTagGroupA && !includeTagGroupB && !includeTagGroupC && !includeTagGroupD) {
+      throw new IllegalArgumentException(
+          "Cannot generate historical logs with no tag groups selected.");
+    }
+
+    // Build string of tag groups for filter type
+    String tagGroupFilterStr = "";
+    if (includeTagGroupA) {
+      tagGroupFilterStr += "A";
+    }
+    if (includeTagGroupB) {
+      tagGroupFilterStr += "B";
+    }
+    if (includeTagGroupC) {
+      tagGroupFilterStr += "C";
+    }
+    if (includeTagGroupD) {
+      tagGroupFilterStr += "D";
+    }
+
+    // Get EBD data type
+    String ebdDataType;
+    if (stringHistorical) {
+      ebdDataType = "HS";
+    } else {
+      ebdDataType = "HL";
+    }
+
+    /*
+     * Build EBD string with parameters
+     * dt[ebdDataType]: data type, value specified in string ebdDataType
+     * ftT: file type, text
+     * startTime: start time for data
+     * endTime: end time for data
+     * flABCD: filter type, specified tag groups
+     */
+    return "$dt"
+        + ebdDataType
+        + "$ftT$st"
+        + startTime
+        + "$et"
+        + endTime
+        + "$fl"
+        + tagGroupFilterStr;
+  }
+
+  /**
+   * Parses Export Block Descriptor Historical Log response into dataPoints. Note: this function
+   * only handles Historical List responses.
+   *
+   * @param exporter EBD Exporter
+   * @return dataPoints
+   * @throws IOException for parsing Exceptions
+   * @throws JSONException for JSON parsing Exceptions
+   */
+  private static ArrayList parseEBDHistoricalLogExportResponse(Exporter exporter)
+      throws IOException, JSONException {
+
+    final String exporterFile = StringUtils.getStringFromInputStream(exporter, "UTF-8");
+    final List eventFileLines = StringUtils.split(exporterFile, "\n");
+    ArrayList dataPoints = new ArrayList();
+    for (int x = 1; x < eventFileLines.size(); x++) {
+      String line = (String) eventFileLines.get(x);
+      DataPoint lineDataPoint = parseHistoricalFileLine(line.trim());
+      if (lineDataPoint != null) {
+        dataPoints.add(lineDataPoint);
+      }
+    }
+
+    exporter.close();
+    return dataPoints;
+  }
+
+  /**
+   * Executes EBD call, waits for data return.
+   *
+   * @param ebdStr EBD string
+   * @return EBD Exporter - caller must close exporter
+   * @throws EbdTimeoutException when there is no response before timeout period
+   * @throws IOException when there is an Exporter Exception
+   */
+  public static Exporter executeEBdCall(String ebdStr) throws IOException, EbdTimeoutException {
```

Comment:
Method name still capitalized incorrectly. B should be lowercase.

Reply:
changed.